### ISSUE TITLE
Is it normal behavior for the Job details page to update the state of "pickup" before assigning an asset?

### ISSUE THREAD

**Issue by faisalgobd** [NerdCats/T-Rex#63]

**Comment by thehoneymad** [comment]
Very good question indeed. And I think this is TaskCat's fault. Im not sure though. But in any case, if you think this is a backend bug, I'd expect a ticket in TaskCat.

@faisalgobd nice work!

**Comment by faisalgobd** [comment]
@thehoneymad  Thanks. I'm not really sure if it needs to come from TaskCat.
We also need to stop user from assigning new asset once the "delivery" state is at "completed" status.
@tareq89 can we manage it from frontend?

**Comment by thehoneymad** [comment]
@faisalgobd yes its possible but the backend is always supposed to know what should and should not be done. So, these should be logged in TaskCat too

**Comment by faisalgobd** [comment]
@thehoneymad I've created ticket at TaskCat. #90

**Comment by tareq89** [comment]
Yes

-----Original Message-----
